### DISCUSSION TITLE
Difference in canvas hash value on same browser

### DISCUSSION THREAD

**Discussion by ravi-ojha** [thumbmarkjs/thumbmarkjs#36] [General]
When clicking the direct link to thumbmarkjs.com from README, it shows a different `commonImageDataHash` than what I get when I copy-paste the link in a new tab. Everything else is same.  Here's the diff [link](https://www.diffchecker.com/5CWzX1Sw/).

Is that a load bug, as in calculating the canvas hash before the full body is loaded? Or a problem with the library?

**Comment by ilkkapeltola** [comment]
Hmm... that's unfortunate. I'm not sure what exactly could be happening here.
I know a way to brute-force this so that it fixes the issue, but it adds unnecessary computation that I would like to avoid.
Nevertheless, we're talking about a few milliseconds so perhaps that would be an acceptable thing for now as a fix.

I'll need to spend some more time on this and see if I can reproduce this locally for debugging. Thanks for reporting though! I'll try to release a patch soon, this is quite important a use-case because the first load on the page needs to be persistent.

**Reply by ilkkapeltola** [reply]
@ravi-ojha which browser are you using? version etc, I'll need to be able to reproduce to have a look.

**Reply by ravi-ojha** [reply]
It's the same as the one from that diff link. `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36` 

Lemme try the updated version and report back.

**Reply by ilkkapeltola** [reply]
Oh right of course. No need, I'll see if I can reproduce when I'm back from vacation.

**Comment by ilkkapeltola** [comment]
In the latest version, I'm doing 3 runs on the canvas and take the common pixels. It doesn't seem to be slow at all, so hope that fixes the immediate issue. Please verify if you could.
